src/models/Validator.ts

```diff
@@ -19,6 +19,9 @@ export class Validator {
         if (Util.isPlayerEvent(message)) {
             const playerEvent: PlayerEvent = message as PlayerEvent;
 
+            if (Util.isRandomCiv(message.civilisation)) {
```

Comment:
I feel like the random civ replacement should not happen inside the Validator, but before

Reply:
I moved it to server.ts